### ISSUE TITLE
Homework 05 is ready for grading 

### ISSUE THREAD

**Issue by yanchaoluo** [yanchaoluo/STAT545-hw-Luo-Yanchao#4]
@vincenzocoia @gvdr @ksedivyhaley @JoeyBernhardt @mynamedaike @pgonzaleze @derekcho

+ [hm05_Luo_Yanchao.md](https://github.com/yanchaoluo/STAT545-hw-Luo-Yanchao/blob/master/hm5/hm05_Luo_Yanchao.md)
+ [hm05 file](https://github.com/yanchaoluo/STAT545-hw-Luo-Yanchao/tree/master/hm5)

SHA: c66c105f3f3098a6fea48ef90c520b2a3aac54b4
Thanks,
Yanchao

**Comment by Jenncscampbell** [comment]
Hi @yanchaoluo

Good job on this homework!! Here is my detailed feedback:

Factor management 
Great job with the factorization comparisons. I did find it a bit hard to follow your r output because the only name distinction is locations or locations3. 
I like how you presented the dropped levels output. This was very clear just to use the nlevels output. I really love the summary chart you made at the end to summarize everything. 
Your `reorder` funcitons were nicely demonstrated. I didn’t know about the `min`, `max` functions that could be specified in the same line!

I was surprised that your `fct_reorder` function didn’t seem to work. It seemed to work okay for me and one difference between our code is I ran a `mutate` function to reorder first which then pipped into my graph. I don’t see why that would make a difference though. One a side note, the x-axis on your graphs is a bit hard to read, I’d suggest using the `theme` function along with an `angle` specification. 

File I/O
I found this part of your assignment to be very detailed! You obviously paid close attention to the changes (for example that sometimes factors get changed to characters). Again your summary chart here was great! This is a really nice reference for future use!


Visualization design
Overall this part is great! I noticed the same legibility issue that I saw above in regard to your axes. 
You explored several features here of different graphing functions like `lumping`. Like you I used this dataset and noticed it was quite hard to graph everything at once since the data becomes cluttered (for example the size differences for familiarity become hard to read). You have a great demo of the `multiplot` function. I actually took a copy of your code for my future use here. I realized here may be a good place to link to a figure file so that you can control the width of these graphs since the output here looks a bit squished. You can actually imbed the graph into your page using `![](http location) 

Writing figures to file
Clean up repo: 
I really loved your table of contents readme at the front of the repo. It was what I used to find your assignment. I did notice that you had a couple repos that are empty or only have a title line so you could probably safely delete those now. 

Progress report: 
For your first `as_factor` problem I think that error line is saying that the function cannot be used on an integer which is what `year` is. You may have to first change it to a character and then to a factor. This might screw up your graphing though. 
For your second issue with the csv file not reading a factor: I think it is actually fine. If you inspect the `artist_name` column there are few in there with these names. I think they may have been data entry errors but the error is not with the csv. It has just reordered the names to be in a different order. 


Overall great job!! I learned some new code from you. I loved your layout. It was very easy to follow your homework. 

Check plus from me! 

Cheers, 

Jenn

**Comment by yanchaoluo** [comment]
Hi @Jenncscampbell 

Thank you so much for an elaborate peer review. Next time I will make the name clearly, and try to use 
`![](http location) ` function to embed the graph. In addition, I would definitely look into `theme` and `angle`  and use them next time. Also, Thank you for pointing my drawbacks out.  😄 
Best regards,

Yanchao

**Comment by wswade2** [comment]
Hi Yanchao,

For your restructuring of the the Singer dataset, I found your code and commentary easy to follow, which is good. The output from your code was not always aesthetically pleasing, but I don't think this is necessarily a bad thing because the output provided important information. Your graphs were likewise informative, but had some aesthetic issues. For example, in many of your plots the x-axis labels are several words jumbled together. When this happens, you may want to either construct your own labels or remove the x-axis labels altogether. I also prefer to see longhand axis labels rather than the shorthand axis labels (e.g. "Artist Name "is better than "artist_name"). I especially appreciated your graph of artist hotness and artist name that used points that grew larger to indicate increaseing artist familiarity. This was a nice touch.

I found that your input/output of some of your data files more than fulfilled what was required by the prompt. You did a good job with this step. You also successfully wrote figures to file and reported your process in a detailed manner

For your repo organization, it did feel like there was a little bit of choice overload when looking at your homework 5 repo. There were so many files that it took me a bit to realize which one I should click on, and you provided 3 links in your read.me. It may be easier for readers if you give them one or two files to choose from.

Wade

**Comment by ksedivyhaley** [comment]
**Factor management (drop & reorder):** Yes
**File I/O (data):** Yes
**Visualization design:** Yes
**File I/O (write figure):** Yes (see comment)
**Organized GitHub:** Partial (see comments)
**Bonus (more forcats, eg relevel):** Yes ( plus fct_lump)
**Reflection:** 

**Comments:**

- For your repo, add “slugs” to the files, eg hw05_FactorsFigures, so you know what is in each assignment (especially since this info is not in your table of contents). Note also that since some of your folders are labelled “h**m**#” and some are “h**w**#” they are out of order. Finally, I recommend using subfolders in each assignment to store saved files (hw05 is a bit crowded).
- “str(singer_locations) shows that no variable is vector.” : Do you mean no variable is a factor?
- “ I found unique of artist_id, artist_name,name are less than total nrows, so I want to factorize artist_id, artist_name,name. ” : Good strategy!
- Good job dealing with the NAs in the dataset.
- “Therefore, if there is no "NA"s as_factor is better than as.factor, because it will keep the data's original order. ” Yes, assuming the data is ordered in the way you want your factor levels to be ordered.
- Nice idea making a summary table for factor management, but watch the length of your code and consider making the table 2x3 for easier reading.
- What happens to the factor levels after you have filtered `year!=0` but before using `droplevels()`?
- Graphs in factor management could be more clearly formatted.
- “There is no difference between dget() and saveRDS() at this time.” There is a difference, but it's in how the data is stored and takes a little more investigation to find.
- Summary table for file I/O is excellent.
- You can embed your figure I/O in the md file instead of just linking to it.
- For bonus, check that you still have factors after using `separate()` - I think this might revert the variables to strings.
- For your first question, have you tried `as.factor` in addition to `as_factor`? 
- For your second question, you are converting the csv character variables into factors using as.factor, which sets factor levels alphabetically. This is why you see weird strings starting with special characters – they come first alphabetically. The “”right output” you were expecting was based on the original order of the data (which is what you see when you set factors using as_factor). 

Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.
